neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/BaseNeo4jDialect.java

```diff
@@ -170,6 +171,20 @@ private void applyMaxRows(QueryParameters queryParameters, StringBuilder nativeQ
 		}
 	}
 
+	private void applyPositionalParameters(QueryParameters queryParameters, StringBuilder nativeQuery) {
+		for ( String key : queryParameters.getNamedParameters().keySet() ) {
+			removePositionalQuestionMark( nativeQuery, key );
+		}
+	}
+
+	private void removePositionalQuestionMark(StringBuilder nativeQuery, String key) {
+		int i = nativeQuery.indexOf( "?" + key );
```

Comment:
To be more precise, when I was talking about this being fragile, I'm thinking about having a `?13` string somewhere in the query and trying to replace `?1`.